### ISSUE TITLE
0install not following HTTP 302 redirects

### ISSUE THREAD

**Issue by bastianeicher** [0install/0install#145]
The version of 0install in Ubuntu 18.04 does not seem to follow HTTP 302 redirects. For example:

```
$ 0install --version
0install (zero-install) 2.12.3

$ 0install download https://apps.0install.net/0install/0watch.xml
Downloaded archive has incorrect size.
URL: https://github.com/0install/0watch/releases/download/0.3.0/0watch-0.3.0.tar.gz
Expected: 1148 bytes
Received: 609 bytes
... downloading implementation https://apps.0install.net/0install/0watch.xml 0.3.0 (id=sha1new=fd6c7fc38284a7ccae1d2713db6844171718da3e)

$ curl -i https://github.com/0install/0watch/releases/download/0.3.0/0watch-0.3.0.tar.gz
HTTP/1.1 302 Found
[...]
location: https://github-production-release-asset-2e65be.s3.amazonaws.com/[...]
[...]
```

The latest generic binaries with cohttp, the old Python implementation and the Windows version all do follow redirects:

```
$ 0install run --version 2.15.1 http://0install.net/tools/0install.xml download https://apps.0install.net/0install/0watch.xml

$ 0install run http://0install.net/2007/interfaces/ZeroInstall.xml https://apps.0install.net/0install/0watch.xml
```

GitHub Releases seems to have started using HTTP 302 redirects recently, causing downloads   of archives hosted on apps.0install.net to fail.

**Comment by talex5** [comment]
This is probably due to the `location` bug fixed in #107 (2.14.1). Older versions ignore a lower-case `location` header, which GitHub seems to be using.

**Comment by bastianeicher** [comment]
Yes, that seems to be it. The 0install build in Ubuntu 19.10 is 2.14.1 and it follows the redirect correctly. Thanks. :)
